jetson/nucleo_bridge/ControllerMap.cpp

```diff
@@ -0,0 +1,94 @@
+#include "ControllerMap.h"
+#include "Controller.h"
+
+//Helper function to calculate an i2c address based off of nucleo # and channel #
+uint8_t ControllerMap::calculate_i2c_address(uint8_t nucleo, uint8_t channel)
+{
+    return ((nucleo + 1) << 4) | channel;
+}
+
+//Helper function to get the path of the config file
+std::string ControllerMap::get_config()
+{
+    std::string configPath = getenv("MROVER_CONFIG");
+    configPath += "/config_nucleo_bridge/controller_config.json";
+    std::ifstream configFile;
+    configFile.open(configPath);
+    
+    std::string config = "";
+    std::string line;
+    while (configFile >> line)
+    {
+        config += line;
+    }
+
+    return config;
+}
+
+//Initialization function
+void ControllerMap::init()
+{
+    rapidjson::Document document;
+    document.Parse(get_config().c_str());
+
+    rapidjson::Value& root = document;
+    assert(root.IsArray());
+
+    for (rapidjson::SizeType i = 0; i < root.Size(); ++i)
+    {
+        assert(root[i].HasMember("name") && root[i]["name"].IsString());
+        std::string name = root[i]["name"].GetString();
+
+        assert(root[i].HasMember("type") && root[i]["type"].IsString());
+        std::string type = root[i]["type"].GetString();
+        
+        assert(root[i].HasMember("nucleo") && root[i]["nucleo"].IsInt());
+        uint8_t nucleo = root[i]["nucleo"].GetInt();
+
+        assert(root[i].HasMember("channel") && root[i]["channel"].IsInt());

```

Comment:
does it just crash if this assert statement is false?

Reply:
Yes. It's ideal because if we get the config file wrong, we likely don't want the rest of the program to run

Reply:
okay, so the indication of that happening is "Initializing virtual controllers\n" gets printed out and then nothing else beyond that (cause it crashes)? can this be documented in the readme?

Reply:
There is a print statement for the initialization of every virtual controller, and assert will print where it failed, so config file issues should be very easy to narrow down. I hope we aren't screwing up the config file too often anyways.

I'll put it in the README